scripts/check_types.py

```diff
@@ -0,0 +1,94 @@
+"""
+Prints a tree with the current status of files with respect to type checking.
+"""
+
+import json
+import logging
+import subprocess
+import sys
+import tempfile
+from collections import defaultdict
+from pathlib import Path
+
+# Set up logging
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+logger = logging.getLogger(__name__)
+
+
+def main():
+    # Setup
+    script_path = Path(__file__).resolve().parent
+    paths = sys.argv[1:] or [
+        script_path.parent / "prolif",
+        script_path.parent / "tests",
+    ]
+    paths = [p for p in paths if Path(p).exists()]
+    if not paths:
+        raise ValueError("Error: No valid paths found")
+    # Find Python files
+    files = []
+    for p in paths:
+        if (path := Path(p)).is_file():
+            if path.suffix == ".py":
+                files.append(path)
+        else:
+            files.extend(path.glob("**/*.py"))
+
+   # Check that there's at least one file to run the type checking on (and log how many files in total) else raise an error
+    logger.info(f"Found {len(files)} Python files")
+    if not files:
+        raise ValueError("No Python files found to run type checking on")
+
+   # Run mypy once
+    logger.info("Running mypy...")
+    with tempfile.NamedTemporaryFile(suffix=".json") as tmp:
+        #Use JSON output
+        subprocess.run(
+            ["mypy", "--disallow-untyped-defs", "--json-report", tmp.name, *paths],
+            check=True, #Check the return code of the process
+            capture_output=True,
+        )
```

Comment:
This command doesn't work currently, the `--json-report` option doesn't exist, closest is `-O json` which prints directly to STDOUT so no need for a tempfile. I'd also remove the `--disallow-untyped-defs` since it's covered by the mypy configuration added in #254. It would also be better to rely on `uv run <command>` to make sure we are using the mypy version in the environment.
So overall change the command to
```suggestion
        result = subprocess.run(
            ["uv", "run", "mypy", "-O", "json", *paths],
            stdout=subprocess.PIPE,
            text=True,
            check=False,
        )
```